spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -652,18 +568,14 @@ private void configureOutboundHandler(AmqpOutboundEndpoint handler, RabbitProper
 	}
 
 	@Override
-	public void bindPubSubProducer(String name, MessageChannel moduleOutputChannel,
-			Properties properties) {
-		validateProducerProperties(name, properties, SUPPORTED_PUBSUB_PRODUCER_PROPERTIES);
+	public void bindProducer(String name, MessageChannel outputChannel, Properties properties) {
+		validateProducerProperties(name, properties, SUPPORTED_PRODUCER_PROPERTIES);
 		RabbitPropertiesAccessor accessor = new RabbitPropertiesAccessor(properties);
 		String exchangeName = applyPrefix(accessor.getPrefix(this.defaultPrefix), name);
```

Comment:
Yes, also we can remove the `doRegisterProducer()` method that accepts a `replyTo` since that is no longer used (request/reply behavior has been removed from the core `Binder` contract).